src/utils/rgbToHSL.ts

```diff
@@ -0,0 +1,44 @@
+/**
+ * Transform RGB to HSL
+ */
+export default function rgbToHSL(r: number, g: number, b: number): number[] {
+  // Make r, g, and b fractions of 1
+  r /= 255;
+  g /= 255;
+  b /= 255;
+
+  // Find greatest and smallest channel values
+  const cmin = Math.min(r, g, b);
+  const cmax = Math.max(r, g, b);
+  const delta = cmax - cmin;
+  let h = 0;
+  let s = 0;
+  let l = 0;
+
+  // Calculate hue
+  // No difference
+  if (delta == 0) h = 0;
+  // Red is max
+  else if (cmax == r) h = ((g - b) / delta) % 6;
+  // Green is max
+  else if (cmax == g) h = (b - r) / delta + 2;
+  // Blue is max
+  else h = (r - g) / delta + 4;
+
+  h = Math.round(h * 60);
+
+  // Make negative hues positive behind 360°
+  if (h < 0) h += 360;
+
+  // Calculate lightness
+  l = (cmax + cmin) / 2;
+
+  // Calculate saturation
+  s = delta == 0 ? 0 : delta / (1 - Math.abs(2 * l - 1));
+
+  // Multiply l and s by 100
+  s = +(s * 100).toFixed(1);
+  l = +(l * 100).toFixed(1);
+
+  return [Math.round(h), Math.round(s), Math.round(l)];
+}
```

Comment:
I dont think that is good to maintain a script to convert RGB to HSL in this repo because is not the focus and we will need to create a lot of unit test to make sure that this works. 

So since you create this solution, I will leave it to you to find out a light lib that converts hue to rgb for us. You can do that?

Reply:
I found a small (27kb) library ([color-convert](https://www.npmjs.com/package/color-convert)) that can convert any number of colour models between each other (`rgb`, `hsl`, `hsv`, `hwb`, `cmyk`, `ansi`, `ansi16`, `hex` strings, and CSS `keyword` s). The API is straightforward and it has a good suite of tests for all conversions.

it might be a good idea to use this library for all color conversions in this package.

I would suggest we use `rgbArray`, `rgbString`, `hslArray`, `hslString`, `hex`, and `keyword` as available options. 

The other options may become useful in future versions of CSS; they would be easy to integrate in the case that they are standardised.

Reply:
I really like this solution. You can go for it. If you need any help let me know